### PR TITLE
New map coloring scheme: system danger

### PR DESCRIPTION
**Pull request by tehhowch**
This was previously semi-bundled in #3070, but I chose to split it out as it makes sense as a standalone feature.
1. This adds a way for the player to discern the relative safety or danger of a given system, beyond the reputation information, as measured by the system's currently-defined fleets and the system government's raid fleet attraction.
2. This does not account for the danger exposed to the player by way of mission NPCs - without asking every mission writer "should the player know exactly where these escorts are?," we cannot determine if the player should know the position of the NPC ships for visible missions. For example, should the player know where the the hostile pirates associated with an `Escort to...` mission are located? My guess is "probably not," so extending the danger calculation to include mission ships is beyond the scope of this PR.
3.  System::Danger does not indicate relative danger, only the absolute. Relative danger would weigh the hostile fleets against those that would help the player in fighting them. Changing this behavior is beyond the scope of this PR.

Example screenshots:

<details><summary>Post Wanderers middle (no raid fleet attraction)</summary>

image
</details>
<details><summary>Post Wanderers middle (12% piracy threat):</summary>

image
</details>
<details><summary>Post Wanderers middle (100% piracy)</summary>

image
</details>
<details><summary>Prior to the Liberate Poisonwood mission:</summary>
(and with Unfettered truce active)

image
</details>
I'm open to changing the names of the items within the legend. I had originally had `"Carefree"` instead of `"None"`, but it didn't work well with the legend headers I tried out.

### CODE REVIEW

**Comment by tehhowch** [comment]
- Rather than piggyback the system name, extending that same art asset (e.g to add a clickable line with the system's danger category) is possible.
 - Is there a better way to represent danger categories? Using the map coloration is definitely superior to the reputation coloration. We could add discrete buckets to mask minor differences while still conveying major differences, but poorly sized buckets could result in a basically useless map. My thought was that the use of categorical words and not numbers in the legend would help prevent knowing the danger at too fine of a detail.

**Comment by Pointedstick** [comment]
To be honest, I'm not sure what this gains the player. A bit of uncertainty is kind of fun, and beyond that, you usually get a grasp pretty quickly of what's dangerous and what's not. Hostile systems will be dangerous, pirate systems will be dangerous, etc.

**Comment by petervdmeer** [comment]
I think "Threat level" sounds more appropriate than "Danger level", but I'm not a native English speaker, so please ignore this remark if you feel Danger level is fine.

I would also make the following changes to the threat-levels:
* None -> Minimal
* Risky -> Gap between Minimal and this one seems large. Maybe add a "Some Risk" step inbetween?
* Uninhabited -> Remove this one (it should still have a level, probably Risky because pirates tend to hang around in uninhabited systems)

Regarding the colors, I would expect Threacherous to be Red. None to be Green and the colors inbetween to nicely flow between those.

When #2678 gets merged, then I would expect this PR to be helpfull for follow-up. since I would expect a follow-up PR where you are only allowed to move escorts forward to "Minimal" systems and they are only allowed to travel through "Minimal" and "Some Risk" systems to get there.

I can also imagine that we will restrict other actions in the game based on the perceived/published risk, for example to only allow ships to be parked in systems with minimal risk. Or by having Dangerous and Treacherous systems spawn hostile ships when player ships are present (holding position or passing through) even when the flagship is elsewhere.

I agree with endless-sky that "Clicking the system name to show the danger level seems really unintuitive.". I would expect this PR to either add an explicit field (below the Pirate name) that shows the systems threat level.

**Comment by jafdy** [comment]
Threat implies that someone is after you, actively trying to get you. The normal lurking pirates are not interested in you specifically so danger is probably a more apt term.

**Review by Amazinite — CHANGES_REQUESTED** [review]
Looks good. I really only have one issue with it, that being the color scheme used. I think the color scheme should be changed to use the blue/light blue/orange/red of the reputation key instead of the blue/light blue/green-ish/yellow of the other keys. Reason being that the reputation map uses red to denote hostility. I would imagine the danger map to do the same, using red to denote the level of hostility in a system. I also have an issue with the current key used including green. I imagine green and blue as both being "friendly" colors, and in fact they are already used as friendly colors in game, with green for your escorts and blue for friendlies. But with this key, the green-ish color is used to denote "dangerous" systems. Basically, the color scheme used currently is dissonant with what it is portraying, using "friendlier" colors to show differences in hostility.

From what I can tell, the "uninhabited" coloring is unused on the danger key. Should uninhabited be used for systems where no fleets ever spawn?

Should the player's fleet strength change what is considered dangerous? Basically meaning that how dangerous a system is has to deal with the strength of the player vs the strength of the enemy instead of just the strength of the enemy. Or does that step into point number 3 in the PR description (which I imagine was referring to friendly fleets that aren't player escorts).
Part of me says "Yes; if a player is stronger than systems become less dangerous," but another part says "No; only worry about the strength of enemy fleets, as that lets you say compare the danger between a pirate system and Mesuket, basically like comparing the levels of enemies in two areas of an RPG; enemies in lower level areas don't lose levels because you gain them, you're just able to survive more dangerous areas, and less dangerous areas become easier."

<hr />

TL;DR
* Request: Change the key colors to shift from blue to red instead of blue to yellow, using red to denote danger.
* Optional: Systems with no fleet spawns show as uninhabited.
* Question for later: Have the player's fleet strength subtract from the danger of a system?

File: source/MapDetailPanel.cpp [hunk 1]
```diff
@@ -223,9 +223,9 @@ bool MapDetailPanel::Click(int x, int y, int clicks)
 			SetCommodity((y - tradeY) / 20);
 			return true;
 		}
-		// Clicking the system name activates the view of the player's reputation with various governments.
+		// Clicking the system name toggles between the systems' danger levels and the player's reputation with their government.
```
[the remark this answers is not in the document]
Comment (Amazinite) [reply source/MapDetailPanel.cpp:226]: Given that we did just add a new help dialog to the map (#5622), I'd suggest adding this toggle behavior to that help dialog.

**Comment by warp-core** [comment]
Is this waiting on tehhowch?

**Comment by Amazinite** [comment]
Closing all PRs older than 2020 which don't have any active discussion. If we wish to revisit these PRs then they may be reopened.
